File: examples/blink.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <string.h>

/* 96BoardsGPIO header file */
#include <gpio.h>

/* Just make the pin numbers easier to remember */
#define GPIO_P1 23

int main(int argc, char * argv[])
{
    int x;
    char board[25];

    if (argc == 2 && (!strcasecmp(argv[1], "dragon") || !strcasecmp(argv[1], "hikey")|| !strcasecmp(argv[1], "bubblegum")) ) {
        strcpy(board,argv[1]);
    }else{
        strcpy(board,"dragon");
    }
    if (!init_96Boards_GPIO_library(board)){
         // Open the GPIO for use.  Do so by pin number on the 
         // Low Speed Expansion Connector.               
         if (!open_GPIO_Board_pin_number(GPIO_P1)){
             // Set the GPIO to output
             if (!setup_GPIO(GPIO_P1, "out")){
                 // Now blink the pin.
                 for (x=0;x<10;x++){
                     digitalWrite(GPIO_P1, HIGH);
                     usleep(500000);
                     digitalWrite(GPIO_P1, LOW);
                     usleep(500000);
                 }
             } else {
                 /* Could not set direction */
                 return(-1);
             }
             // Now close the GPIO used
             close_GPIO(GPIO_P1);
         } else {
             /* Could not open GPIO pin */
             return(-1);
         }
     }else{
         /* Could not initi library */
         return(-1);
     }
     // Exit the example program
     return 0;
}

File: examples/blink.py
#!/usr/bin/python
import time

from gpio_96boards import GPIO

GPIO_P1 = 23
pins = (
    (GPIO_P1, 'out'),
)


def blink(gpio):
    for i in range(5):
        gpio.digital_write(GPIO_P1, GPIO.HIGH)
        time.sleep(i)
        gpio.digital_write(GPIO_P1, GPIO.LOW)
        time.sleep(1)


if __name__ == '__main__':
    import argparse
    import os

    mapping_file_exists = os.path.exists('/etc/96boards_gpio.conf')
    parser = argparse.ArgumentParser(
        description='Blink LED on GPIO 1 (pin 23)')
    if not mapping_file_exists:
        parser.add_argument('board', help='Name of board. eg dragon')
    args = parser.parse_args()
    if mapping_file_exists:
        args.board = ''

    with GPIO(args.board, pins) as gpio:
        blink(gpio)
